### PR TITLE
Pocket open source project

### PR DESCRIPTION
**Pull request by CMasterson**
## Summary
Make Pocket iOS more welcoming to outside contributors.
Add missing secrets examples
Fix a crash that happens with an invalid GroupID

### CODE REVIEW

File: PocketKit/Sources/Sync/PersistentContainer.swift [hunk 1]
```diff
@@ -36,21 +36,7 @@ public class PersistentContainer: NSPersistentContainer {
...
-                .containerURL(forSecurityApplicationGroupIdentifier: groupID)!
-                .appendingPathComponent("PocketModel.sqlite")
-
-            Log.debug("Store URL: \(sharedContainerURL)")
-            storeDescription = NSPersistentStoreDescription(url: sharedContainerURL)
-        }
-
-        guard let storeDescription else {
-            fatalError("no store description")
-        }
```
Comment (Gio2018) [inline PocketKit/Sources/Sync/PersistentContainer.swift:53]: Not sure if I'm getting this right, but with this change, it seems that there is a path where, if we initialize with the `shared` storage and, for some reason, the on disk db is not available, we' re still initializing the app with a volatile storage. Is this what we want? cc @bassrock

File: PocketKit/Sources/Sync/PersistentContainer.swift [hunk 2]
```diff
@@ -15,7 +15,7 @@ public class PersistentContainer: NSPersistentContainer {
     }()
 
     private (set) var spotlightIndexer: CoreDataSpotlightDelegate?
-    private (set) var storeDescription: NSPersistentStoreDescription?
+    private (set) var storeDescription: NSPersistentStoreDescription!
```
Comment (Gio2018) [inline PocketKit/Sources/Sync/PersistentContainer.swift:18]: what's the advantage of implicitly unwrapping here?

**Review by Gio2018 — COMMENTED** [review]
Had a couple of comments @CMasterson

File: PocketKit/Sources/Sync/PersistentContainer.swift (hunk 2 above)
Comment (CMasterson) [reply PocketKit/Sources/Sync/PersistentContainer.swift:18]: `getStoreDescription` returns non-optional.
So I removed the optionalness of `storeDescription` and xcode complained about not all properties being initalised before the class called `super.init`.
`storeDescription` will be valid before it's accessed or something horrifying has happened and so we should probably crash out.

File: PocketKit/Sources/Sync/PersistentContainer.swift (hunk 1 above)
Comment (CMasterson) [reply PocketKit/Sources/Sync/PersistentContainer.swift:53]: The scenario I coded for was if we ask for .shared storage with a bad groupID, we fall back to .inMemory. Maybe we need a more explicit `if groupID is the default garbage value { fall back to .inMemory }` but that seems specific and brittle.

Comment (Gio2018) [reply PocketKit/Sources/Sync/PersistentContainer.swift:53]: moving a bit of conversation to slack on this

**Comment by CMasterson** [comment]
Closing this as I have been approved to add GroupID to the default secrets, making this change unnecessary.

**Pull request closed without merging**